tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -116,3 +167,85 @@ def test_band_data(self):
         self.assertEqual(self.band.min, 0)
         self.assertEqual(self.band.max, 255)
         self.assertEqual(self.band.nodata_value, 15)
+
+    def test_read_mode_error(self):
+        # Open raster in read mode
+        rs = GDALRaster(self.rs_path, write=False)
+        band = rs.bands[0]
+
+        # Setting attributes in write mode raises exception in the _flush method
+        self.assertRaises(GDALException, setattr, band, 'nodata_value', 10)
+
+    def test_band_data_setters(self):
+        # Create in-memory raster and get band
+        rsmem = GDALRaster({'datatype': 1, 'driver': 'MEM',
+                            'name': 'mem_rst', 'width': 10, 'height': 10, 'nr_of_bands': 1})
+        bandmem = rsmem.bands[0]
+
+        # Set nodata value
+        bandmem.nodata_value = 99
+        self.assertEqual(bandmem.nodata_value, 99)
+
+        # Set data for entire dataset
+        bandmem.data(range(100))
+        if numpy:
+            self.assertTrue((bandmem.data() ==
```

Comment:
I came across `numpy.testing.assert_array_equal`. Maybe it would be worth using that.